custom_components/bond/light.py

```diff
@@ -22,71 +31,191 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
+                                          deviceId,
+                                          device,
+                                          deviceProperties,
+                                          supportsFlameActions)
+                add_entities([fireplace])
 
-        # If the device type is not Ceiling Fan, or it is Ceiling Fan but has no action for light control
-        # then don't create a light instance
-        if newBondLight._properties['type'] == BOND_DEVICE_TYPE_CEILING_FAN and \
-              ( BOND_DEVICE_ACTION_TURNLIGHTON in newBondLight._properties['actions'] or \
-                BOND_DEVICE_ACTION_TURNLIGHTOFF in newBondLight._properties['actions'] or \
-                BOND_DEVICE_ACTION_TOGGLELIGHT in newBondLight._properties['actions'] ):
-            add_entities( [ newBondLight ] )
 
 class BondLight(Light):
     """Representation of an Bond Light."""
 
-    def __init__(self, bond, deviceId):
+    def __init__(self, bond, deviceId, device, properties):
         """Initialize a Bond Light."""
         self._bond = bond
         self._deviceId = deviceId
-        self._properties = self._bond.getDevice(self._deviceId)
-        self._name = self._properties['name'] + ' Light'
+        self._device = device
+        self._properties = properties
+        name = "Light" if "name" not in properties else properties['name']
+        if "location" in properties:
+            self._name = f"{properties['location']} {name}"
+        else:
+            self._name = name
         self._state = None
 
     @property
     def name(self):
         """Return the display name of this light."""
         return self._name
 
-    # @property
-    # def brightness(self):
-    #     """Return the brightness of the light.
+    @property
+    def is_on(self):
+        """Return true if light is on."""
+        return self._state
 
-    #     This method is optional. Removing it indicates to Home Assistant
-    #     that brightness is not supported for this light.
-    #     """
-    #     return self._brightness
+    def turn_on(self, **kwargs):
+        """Instruct the light to turn on."""
+        self._bond.turnLightOn(self._deviceId)
+
+    def turn_off(self, **kwargs):
+        """Instruct the light to turn off."""
+        self._bond.turnLightOff(self._deviceId)
+
+    def update(self):
+        """Fetch new state data for this light.
+        This is the only method that should fetch new data for Home Assistant.
+        """
+        bondState = self._bond.getDeviceState(self._deviceId)
+        if 'light' in bondState:
+            self._state = True if bondState['light'] == 1 else False
+
+    @property
+    def unique_id(self):
+        """Get the unique identifier of the device."""
+        return self._deviceId
+
+    @property
+    def device_id(self):
+        """Return the ID of this light."""
+        return self.unique_id
+
+
+class BondFireplace(Light):
+    """Representation of an Bond Fireplace."""
```

Comment:
I guess this makes sense to treat a fireplace like a light. I get it, with the dimmer and all... it's just kinda funny to think about. I guess it does emit light...

Reply:
Yeah, I thought the same. There really is no other entity type that fits in here though :\